templates/firstboot/mlnx_fw_update.py

```diff
@@ -89,16 +91,21 @@ def discover(self):
             return self._devs
 
         devs = []
-        cmd = ['lspci', '-d', '15b3:']
+        cmd = ['lspci', '-D', '-d', '15b3:']
         out = run_command(*cmd)
         for line in out.split('\n'):
             if not line:
                 continue
             dev = line.split()[0]
-            if dev.endswith('.0'):
+            if dev.endswith('.0') and (not self._dev_white_list or
+                    dev in self._dev_white_list):
                 devs.append(dev)
         self._devs = devs
         LOG.info("Found Mellanox devices: %s", devs)
+        other_devs = set(self._dev_white_list) - set(devs)
+        if other_devs:
+             LOG.warning("Uproccessed devices %s which may be not Mellanox devices, "
```

Comment:
this message is unclear to me.  perhaps something like:

"Not all devices in PCI white list where discovered, these may not be mellanox devices or have their PCI function set to non zero. "